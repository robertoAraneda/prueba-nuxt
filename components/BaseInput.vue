<script setup lang="ts">

import { useAttrs } from 'vue';

const model = defineModel<string | number>();

</script>

<template>
  <v-text-field v-model="model"></v-text-field>
</template>