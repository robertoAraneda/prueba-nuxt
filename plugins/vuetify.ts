// import this after install `@mdi/font` package
import vuetifyConfig from "../vuetify.config"

export default defineNuxtPlugin((app) => {
  app.vueApp.use(vuetifyConfig)
})