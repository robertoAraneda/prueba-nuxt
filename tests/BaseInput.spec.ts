import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import BaseInput from '~/components/BaseInput.vue'
import vuetify from '~/vuetify.config';


describe("BaseInput", () => {
  it("should mount a component", async () => {
    expect(BaseInput).toBeTruthy();
  });

  it("should render a component", async () => {
    const wrapper = mount(BaseInput, {
      global: {
        plugins: [vuetify]
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a component with a label", async () => {
    const wrapper = mount(BaseInput, {
      global: {
        plugins: [vuetify]
      },
  
      props: {
        label: "Username",
        
      }
    });

    const input = wrapper.find("input");
    expect(input.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  }
  );

});