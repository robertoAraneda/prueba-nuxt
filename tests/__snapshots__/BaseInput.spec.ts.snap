// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseInput > should render a component 1`] = `
"<div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field">
  <!---->
  <div class="v-input__control">
    <div class="v-field v-field--center-affix v-field--no-label v-field--variant-filled v-theme--light v-locale--is-ltr">
      <div class="v-field__overlay"></div>
      <div class="v-field__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="opacity: NaN;"></div>
          <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div class="v-field__field" data-no-activator="">
        <!----><label class="v-label v-field-label" for="input-0">
          <!---->
          <!---->
        </label>
        <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" class="v-field__input">
        <!---->
      </div>
      <!---->
      <!---->
      <div class="v-field__outline">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <div class="v-input__details">
    <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
      <!---->
    </transition-group-stub>
    <!---->
  </div>
</div>"
`;

exports[`BaseInput > should render a component with a label 1`] = `
"<div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field">
  <!---->
  <div class="v-input__control">
    <div class="v-field v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
      <div class="v-field__overlay"></div>
      <div class="v-field__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="opacity: NaN;"></div>
          <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-0">
          <!---->Username
        </label><label class="v-label v-field-label" for="input-0">
          <!---->Username
        </label>
        <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" class="v-field__input">
        <!---->
      </div>
      <!---->
      <!---->
      <div class="v-field__outline">
        <!---->
        <!---->
      </div>
    </div>
  </div>
  <!---->
  <div class="v-input__details">
    <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
      <!---->
    </transition-group-stub>
    <!---->
  </div>
</div>"
`;
